src/Storages/System/StorageSystemDroppedTablesParts.cpp

```diff
@@ -0,0 +1,303 @@
+#include <Storages/System/StorageSystemDroppedTablesParts.h>
+#include <atomic>
+#include <memory>
+#include <string_view>
+
+#include <Storages/MergeTree/MergeTreeData.h>
+#include <DataTypes/DataTypeString.h>
+#include <DataTypes/DataTypeArray.h>
+#include <DataTypes/DataTypesNumber.h>
+#include <DataTypes/DataTypeDateTime.h>
+#include <DataTypes/DataTypeDate.h>
+#include <DataTypes/DataTypeUUID.h>
+#include <Parsers/queryToString.h>
+#include <Interpreters/TransactionVersionMetadata.h>
+#include <Interpreters/Context.h>
+
+
+namespace
+{
+
+std::string_view getRemovalStateDescription(DB::DataPartRemovalState state)
```

Comment:
Maybe move it closer to a definition of the enum itself to avoid code duplication?